### PR TITLE
Feat/config file parsing

### PR DESCRIPTION
**Pull request by AJMCoding**
### Description

Added some config files + a class that is able to parse a configfile data into a struct.

### Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Documentation update

### How to Test

by modifing the configfile_example (located in the config_files folder) or by calling another config file by writing the file name as an input. (./webserv <file_name>)

### Checklist

- [x] Code follows project style guidelines
- [x] Documentation has been updated

### CODE REVIEW

File: src/configfile/Parser_helpers.cpp [hunk 1]
```diff
@@ -0,0 +1,592 @@
...
+  methods.push_back("PROPPATCH");
+  methods.push_back("LOCK");
+  methods.push_back("UNLOCK");
+  methods.push_back("PATCH");
+  std::vector<std::string> allowed_methods;
+  for (unsigned long i = 1; i < args.size(); i++) {
+    for (unsigned long j = 0; j < methods.size(); j++) {
+      if (args[i] == methods[j]) {
+        if (args[i] == "GET") {
+          allowed_methods.push_back(args[i]);
```
Comment (Jakobus42) [inline src/configfile/Parser_helpers.cpp:319]: I am confused

File: src/configfile/Parser_helpers.cpp [hunk 2]
```diff
@@ -0,0 +1,592 @@
...
+  else if (key == "root")
+    i = root(args, line_count, 2);
+  else if (key == "autoindex")
+    i = autoindex(args, line_count, 2);
+  else if (key == "index")
+    i = index(args, line_count, 2);
+  else
+    i = 2;
+  return i;
+}
```
Comment (Jakobus42) [inline src/configfile/Parser_helpers.cpp:590]: This solution seems like a work around to test pvt member functions. I am sure there is a better way

File: include/configfile/Parser.hpp [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <sstream>
+#include <vector>
+
+#include "../../include/configfile/Parser.hpp"
+#include "../../include/logger/LoggerManager.hpp"
+struct location {
```
Comment (Jakobus42) [inline include/configfile/Parser.hpp:11]: why is this not a typedef?

File: src/configfile/Parser_helpers.cpp [hunk 4]
```diff
@@ -0,0 +1,592 @@
...
+  }
+  return temp;
+}
+
+/**
+ * @brief Converts a string to a positive integer.
+ * @param str The string to convert.
+ * @return int The integer value of the string. or -1 if the string is not a number.
+ */
+int ft_stoi(std::string str) {
```
Comment (Jakobus42) [inline src/configfile/Parser_helpers.cpp:35]: Why not use stringstream instead of writing your own stoi?

File: src/configfile/Parser.cpp [hunk 5]
```diff
@@ -0,0 +1,366 @@
...
+ * @brief Copy constructor.
+ * @param other The other ConfigFileParser object to copy.
+ */
+ConfigFileParser::ConfigFileParser(const ConfigFileParser&) { m_isLoaded = 0; }
+
+/* *
+ * @brief Constructs a new ConfigFileParser object.
+ * @param configFileName The name of the configuration file.
+ */
+/* ConfigFileParser::ConfigFileParser(std::string &configFileName)
```
Comment (Jakobus42) [inline src/configfile/Parser.cpp:24]: Please dont submit commented out code.

**Review by Jakobus42 — CHANGES_REQUESTED** [review]
Hey:) Thank you for your first PR.
In general the config parser looks fairly okay but I still found some issues I would be happy to be fixed.

I would suggest to abstract and group common  functionality.
It also would make sense to go over the code again and try to make it simpler / more readable.
What I mean by this is separating bigger functions into smaller ones. 

Also you should try to not use so many "magic values" try to use enums and constants for those cases.

Please name your module "config" instead of config parser. Aswell as the namespace and the class.
Also move your config_files into the config folder with the existing 2 example config files.

**Comment by rwxg** [comment]
I'll be honest, this code makes my head spin. As long as it truly works, reliably and as expected, and it remains a black box that runs exactly once at startup, I'm _fine_ with this staying as-is.

**Review by Jakobus42 — DISMISSED** [review]

**Review by rwxg — APPROVED** [review]
LGTM

**Pull request merged**